driver/network_driver.go

```diff
@@ -252,8 +290,12 @@ func (d NetworkDriver) Join(request *network.JoinRequest) (*network.JoinResponse
 		return nil, err
 	}
 
+	mac, err := d.GetEndpointMAC(request.EndpointID)
+	if err != nil {
+		return nil, err
+	}
 	// libnetwork doesn't set the MAC address properly, so set it here.
```

Comment:
Done in https://github.com/projectcalico/libnetwork-plugin/pull/129/commits/b1b6b033b53c331c8f521fd709369b097098d81f